File: src/main/java/com/example/CargoTracking/model/DomesticShipment.java
package com.example.CargoTracking.model;

import lombok.*;
import org.springframework.format.annotation.DateTimeFormat;
import org.springframework.web.multipart.MultipartFile;

import javax.persistence.*;
import java.sql.Date;
import java.sql.Time;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.time.LocalTime;
import java.util.List;


@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@ToString
@Data
@Builder
@Entity
public class DomesticShipment {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private int trip;
    private String originCountry;
    private String destinationCountry;
    private String originFacility;
    private String originLocation;
    private Boolean refrigeratedTruck;//vehicle
    private String destinationFacility;
    private String destinationLocation;
    private Long originLocationId;
    private Long destinationLocationId;
    private String routeNumber;
    private Long routeNumberId;
    private Integer numberOfShipments;
//    private Double weight;//weight in kg
    private LocalDateTime etd;
    private LocalDateTime eta;
    private LocalDateTime atd;
    private String driverName;
    private String driverContact;
    private String referenceNumber;//Master CONS
    private String vehicleType;
    private Integer numberOfPallets;
    private Integer numberOfBags;//Number of bags non document
    private Integer numberOfPalletsReceived;
    private Integer numberOfBagsReceived;
    private String vehicleNumber;
    private String tagNumber;//security tag
    private Long sealNumber;
    private Integer numberOfBoxes;//Number of bags document
    private String status;
    private String remarks;
    private LocalDateTime ata;
    private Integer totalShipments;
    private Integer overages;
    private String overagesAwbs;
    private Integer received;
    private Integer shortages;
    private String shortagesAwbs;
    private String attachments;
    private Boolean redFlag;
    @Column(unique = true)
    private String preAlertNumber;
    private String preAlertType;
    private long transitTimeTaken;
    private LocalDateTime createdTime;
    private LocalDateTime updatedTime;
    private LocalDateTime arrivedTime;
    private LocalDateTime clearedTime;
    private Boolean activeStatus;
    private boolean escalationFlagOne;
    private boolean escalationFlagTwo;
    private boolean escalationFlagThree;
    private LocalDate createdAt;
    private LocalDate updatedAt;
    private Integer damage;
    private String damageAwbs;
    @ManyToOne
    private User createdBy;
    @ManyToOne
    private User UpdatedBy;

    @OneToMany(mappedBy = "domesticShipment",cascade = CascadeType.REMOVE)
    private List<DomesticShipmentHistory> domesticShipmentHistories ;

    @OneToMany(mappedBy = "domesticShipment", cascade = CascadeType.REMOVE)
    private List<FileMetaData> files;

    @Override
    public String toString() {
        return "DomesticShipment{" +
                "id=" + id +
                ", originFacility='" + originFacility + '\'' +
                ", originLocation='" + originLocation + '\'' +
                ", refrigeratedTruck=" + refrigeratedTruck +
                '}';
    }
}

File: frontend/src/app/model/DomesticShipment.ts
import { InputNumberInputEvent } from "primeng/inputnumber";

export interface DomesticShipment{
    originCountry: string|null|undefined,
    destinationCountry: string|null|undefined,
    originFacility: string|null|undefined,
    originLocation: string|null|undefined,
    refrigeratedTruck: boolean,//Refrigerated Vehicle
    destinationFacility: string|null|undefined,
    destinationLocation: string|null|undefined,
    routeNumber: string|null|undefined,
    numberOfShipments: number|null|undefined,
    weight: number|null|undefined,
    preAlertNumber:string|null|undefined,
    // etd: Date|string|null|undefined,
    // eta: Date|string|null|undefined,
    atd: Date|string|null|undefined,
    driverName: string|null|undefined,
    driverContact: string|null|undefined,
    referenceNumber: string|null|undefined,//Master CONS
    vehicleType: string|null|undefined,
    numberOfPallets: number|null|undefined,
    numberOfBags: number|null|undefined,
    vehicleNumber: string|null|undefined,
    tagNumber: string[]|string|null;//security tag
    // sealNumber: number|null|undefined,
    routeNumberId:number|null|undefined
    status: string|null|undefined,
    remarks: string|null|undefined,
    ata: Date|string|null|undefined,
    totalShipments: number|null|undefined,
    overages: number|null|undefined,
    overagesAwbs: string|null|undefined,
    numberOfBoxes:number|null|undefined,
    received: number|null|undefined,
    shortages: number|null|undefined,
    shortagesAwbs: string|null|undefined,
    attachments: string|null|undefined,
    transitTimeTaken:number|null,
    preAlertType:string|null,
    damage:number|null,
    damageAwbs:string|null,
    numberOfPalletsReceived:number|null,
    numberOfBagsReceived:number|null;
}
